test/e2e/ingress.go

```diff
@@ -224,16 +222,166 @@ func kubectlLogLBController(c *client.Client) {
 	}
 	for _, p := range podList.Items {
 		Logf("\nLast 100 log lines of %v\n", p.Name)
-		l, _ := runKubectl("logs", p.Name, fmt.Sprintf("--namespace=%v", api.NamespaceSystem), "-c", lbContainerName, "--tail=100")
+		l, _ := runKubectl("logs", p.Name, fmt.Sprintf("--namespace=%v", ns), "-c", lbContainerName, "--tail=100")
 		Logf(l)
 	}
 }
 
+type ingressController struct {
+	ns             string
+	rcPath         string
+	defaultSvcPath string
+	UID            string
+	rc             *api.ReplicationController
+	svc            *api.Service
+	c              *client.Client
+}
+
+func (cont *ingressController) create() {
+
+	// TODO: This cop out is because it would be *more* brittle to duplicate all
+	// the name construction logic from the controller cross-repo. We will not
+	// need to be so paranoid about leaked resources once we figure out a solution
+	// for issues like #16337. Currently, all names should fall within 63 chars.
```

Comment:
Yeah we can lose that.

To clarify, the name namespace currently is 'e2e-test-glbc-5chars', which won't cause problems. The 5 chars limit is from the master (ingress strategy). If we make this arbitrarily long, it will, and the right way to solve this is by asking the controller what resources it created before deleting the Ingress, then checking that those exact resources are deleted. 

I should probably add that in the comment in case it starts to fail for this reason when I'm no longer around or something.